services/director/src/simcore_service_director/registry_proxy.py

```diff
@@ -1,25 +1,70 @@
 #pylint: disable=C0111
+import asyncio
 import enum
 import json
 import logging
 from typing import Dict, List, Tuple
 
 import aiohttp
+from aiohttp import web
 from yarl import URL
 
 from . import config, exceptions
 
 DEPENDENCIES_LABEL_KEY = 'simcore.service.dependencies'
 
-NUMBER_OF_RETRIEVED_REPOS = 5
-NUMBER_OF_RETRIEVED_TAGS = 5
+NUMBER_OF_RETRIEVED_REPOS = 50
+NUMBER_OF_RETRIEVED_TAGS = 50
 
 _logger = logging.getLogger(__name__)
+_registry_requests_cache = {}
+TASK_NAME = __name__ + "_registry_caching_task"
+TASK_STATE = "{}_state".format(TASK_NAME)
 
 class ServiceType(enum.Enum):
+    ALL = ""
     COMPUTATIONAL = "comp"
     DYNAMIC = "dynamic"
 
+class TaskState(enum.IntEnum):
+    STARTING = 0
+    RUNNING = 1
+    FAILED = 2
+    STOPPED = 3
+
+async def registry_caching_task(app: web.Application):
+    _logger.info("starting registry caching task")
```

Comment:
what you call a "caching task"  is in reality a cache cleaner ... this task does not cache anything. i would rename it for clarity

Reply:
that is not true. the call to list_services repopulates the cache.

Reply:
aha! my bad! :-)